### PR TITLE
Move NodeId Generation to a separate crate

### PR DESCRIPTION
**Pull request by am357**
*Description of changes:*

Moves the `NodeId` and its Id generation to a new crate, `partiql-common` so that it can get consumed by multiple crates.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: partiql-ast/src/ast.rs [hunk 1]
```diff
@@ -17,13 +17,10 @@ use std::fmt;
 use serde::{Deserialize, Serialize};
 
 use partiql_ast_macros::Visit;
+use partiql_core::node::NodeId;
 
 pub type AstTypeMap<T> = IndexMap<NodeId, T>;
```
Comment (jpschorr) [inline partiql-ast/src/ast.rs:22]: Should this move and rename as well?

Comment (jpschorr) [reply partiql-ast/src/ast.rs:22]: If it does, then `partiql-source-map` could stop importing `ast`.

**Review by jpschorr — COMMENTED** [review]
I wonder if there is a better name than `partiql-core`?

Perhaps we could move this type and `partiql-source-map` somewhere...

File: partiql-ast/src/ast.rs (hunk 1 above)
Comment (am357) [reply partiql-ast/src/ast.rs:22]: Makes sense; could be `NodeTypeMap` or something similar.

**Comment by am357** [comment]
> I wonder if there is a better name than `partiql-core`?
Not sure TBH—happy to change if you have a suggestion.

> Perhaps we could move this type and `partiql-source-map` somewhere...
I could see `partiql-source-map` going under: `<partiql-core>/src/syntax-source/...`. Thoughts?

File: partiql-ast/src/ast.rs (hunk 1 above)
Comment (am357) [reply partiql-ast/src/ast.rs:22]: > Makes sense; could be `NodeTypeMap` or something similar.

Just realized I have already moved but haven't removed this one:
`pub type NodeMap<T> = IndexMap<NodeId, T>;`

**Comment by jpschorr** [comment]
> > I wonder if there is a better name than `partiql-core`?
> 
> Not sure TBH—happy to change if you have a suggestion.

We talked out-of-band and thing `partiql-common` may be less ambiguous.


> > Perhaps we could move this type and `partiql-source-map` somewhere...
> 
> I could see `partiql-source-map` going under: `<partiql-core>/src/syntax-source/...`. Thoughts?

Sure

**Review by jpschorr — APPROVED** [review]

**Pull request merged**